ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/MethodModel.java

```diff
@@ -231,13 +236,17 @@ private Object synchronousCall(Invocation.Builder builder, Object entity, Method
         return response.readEntity(returnType);
     }
 
-    private CompletableFuture asynchronousCall(Invocation.Builder builder, Object entity, Method method) {
+    private CompletableFuture asynchronousCall(Invocation.Builder builder,
+                                               Object entity,
+                                               Method method,
+                                               MultivaluedMap<String, Object> customHeaders) {
         CompletableFuture<Object> result = new CompletableFuture<>();
         Future<Response> theFuture;
         if (entity != null
                 && !httpMethod.equals(GET.class.getSimpleName())
                 && !httpMethod.equals(DELETE.class.getSimpleName())) {
-            theFuture = builder.async().method(httpMethod, Entity.entity(entity, consumes[0]));
+            String contentType = (String) customHeaders.get(HttpHeaders.CONTENT_TYPE).get(0);
```

Comment:
Duplication of code - maybe a new method to get `contentType` from the customHeaders map?